src/Components/Aspire.Azure.Messaging.EventHubs/EventProcessorClientComponent.cs

```diff
@@ -0,0 +1,93 @@
+        string connectionName, string configurationSectionName)
+    {
+        return azureFactoryBuilder.RegisterClientFactory<EventProcessorClient, EventProcessorClientOptions>(
+            (options, cred) =>
+            {
+                var connectionString = settings.ConnectionString;
+                if (string.IsNullOrEmpty(connectionString) && string.IsNullOrEmpty(settings.Namespace))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventHubProducerClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'ConnectionString' or 'Namespace' in the '{configurationSectionName}' configuration section.");
+                }
+
+                if (string.IsNullOrEmpty(settings.EventHubName))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventProcessorClient could not be configured. Ensure a valid EventHubName was provided in the '{configurationSectionName}' configuration section.");
+                }
+
+                options.Identifier ??= GenerateClientIdentifier(settings);
+
+                var blobClient = GetBlobContainerClient(settings, cred, configurationSectionName);
+
+                var processor = !string.IsNullOrEmpty(connectionString)
+                    ? new EventProcessorClient(blobClient,
+                        settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName, connectionString)
+                    : new EventProcessorClient(blobClient,
+                        settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName, settings.Namespace,
+                        settings.EventHubName, cred, options);
+
+                return processor;
+
+            }, requiresCredential: false);
+    }
+
+    private BlobContainerClient GetBlobContainerClient(
+        AzureMessagingEventHubsSettings settings, TokenCredential cred, string configurationSectionName)
+    {
+        if (string.IsNullOrEmpty(settings.BlobClientConnectionName))
+        {
+            // throw an invalid operation exception if the blob client connection name is not provided
+            throw new InvalidOperationException(
+                $"A EventProcessorClient could not be configured. Ensure a valid connection name was provided in the '{configurationSectionName}' configuration section.");
+        }
+
+        var blobConnectionString =
+            builderConfiguration.GetConnectionString(
+                settings.BlobClientConnectionName) ??
+                throw new InvalidOperationException(
+                    "A EventProcessorClient could not be configured. " +
+                    $"There is no connection string in Configuration with the name {settings.BlobClientConnectionName}. " +
+                    "Are you sure you defined a Blob resource with this name in your AppHost?");
```

Comment:
This message is kind of weird. These components can be used outside of an Aspire AppHost. Maybe word to it to say "Ensure you have configured an Azure Storage Blob connection ....".

Reply:
Good idea.